### ISSUE TITLE
Discriminator and Spectrogram Collector should enforce fMin(Max)Frequency bounds in conjunction with frequency bounds from a track

### ISSUE THREAD

**Issue by evzayas** [project8/katydid#76]
Currently these are the only two processors with a minimum and maximum frequency that can be determined/set by incoming data (i.e. a track), or by the configuration values `min(max)-frequency`. The former is implemented in a way such that it ignores the latter; for example, I might tell the spectrogram collector to set the minimum frequency 10 MHz below the track starting frequency, and simultaneously specify an overall minimum frequency of 50 MHz:

```
spectrogram:
    min-frequency: 50e6
    max-frequency: 150e6
    lead-freq: 10e6
    trail-freq: 10e6
    use-track-freqs: true
```

If a track is received with starting frequency 55 MHz, the minimum frequency will be set to 45 MHz even though `min-frequency` has been specified above that. The logic with this should be changed to simply take the greater (lesser) of the two minima (maxima), so that neither bound is violated.

**Comment by evzayas** [comment]
This is now fixed in the feature/LineFitting branch. I plan to merge this branch into develop at some point soon, so I will not do a hotfix.

Commit is here: [dffef85](https://github.com/project8/katydid/commit/dffef859248bb88157c9a70f5926c3587736c769)

**Comment by evzayas** [comment]
I will also close the issue once the merge to develop is done, presumably next week.

**Comment by evzayas** [comment]
Branch is now merged into develop. Closing the issue
